### DISCUSSION TITLE
Is there a document outlining Tock's philosophy for implementing wireless protocols?

### DISCUSSION THREAD

**Discussion by potto216** [tock/tock#3357] [Q&A] [answered]
Wireless protocols like Bluetooth LE, 6LowPAN, Threads, and ZigBee are complex and evolving. Additionally, hardware implementing the radios is varied and specialized. Yet, these protocols are becoming increasingly common communication paths for devices with embedded operating systems. @hudson-ayers [highlighted the issues for implementing Thread](https://github.com/tock/tock/discussions/3193#discussioncomment-3565347) and protocols that require using encryption hardware.  When I looked at Tock's implementation of  6LowPAN, UDP, and Bluetooth LE advertising for the nRF52 I can appreciate the large amount of hours that went into these efforts. Given finite resources is there a document outlining Tock's philosophy for deciding what wireless protocols to implement and how?

**Comment by hudson-ayers** [comment, marked as answer]
I think the closest document to what you are describing is the "principles" section of [this doc](https://github.com/tock/tock/blob/master/doc/Networking_Stack.md#principles). The rest of that document is somewhat out of date -- it detailed a roadmap to a full Thread implementation, but we never got that far. It still offers a decent window into the philosophies applied the last time we made a significant networking push in Tock, though

**Reply by potto216** [reply]
Thanks! What is the best way to have discussions about the document?

**Reply by hudson-ayers** [reply]
There are these two tracking issues: https://github.com/tock/tock/issues/3308 and https://github.com/tock/tock/issues/983 . One of those might be appropriate, depending on what you want to discuss. Feel free to open a new issue as well if your discussion doesn't closely fit the topic of either of those issues.

**Comment by bradjc** [comment]
We now have a networking working group as well: https://github.com/tock/tock/tree/master/doc/wg/network
